docs/guide/README.md

```diff
@@ -0,0 +1,120 @@
+# User Guide
```

Comment:
The documentation is clear, a high-level summary table at the top would be beneficial. It should list the supported data types and callable types for all methods in the https://github.com/TRAIS-Lab/dattri?tab=readme-ov-file#supported-algorithms.